app/src/main/java/com/example/moviews/data/local/MovieLocalDataSource.kt

```diff
@@ -0,0 +1,29 @@
+package com.example.moviews.data.local
+
+import com.example.moviews.data.MovieDataSource
+import com.example.moviews.data.OnLoadDataCallback
+import com.example.moviews.data.local.dao.MovieDao
+import com.example.moviews.data.model.Movie
+
+class MovieLocalDataSource private constructor(
+    private val movieDao: MovieDao
+) : MovieDataSource.Local {
+
+    override fun insertMovie(movie: Movie, callback: OnLoadDataCallback<Boolean>) {
+        LocalAsyncTask<Movie, Boolean>(callback) {
+            movieDao.insertMovie(movie)
+        }.execute(movie)
+    }
+
+    override fun deleteMovie(movieId: String, callback: OnLoadDataCallback<Boolean>) {
+        LocalAsyncTask<String, Boolean>(callback) {
+            movieDao.deleteMovie(movieId)
+        }.execute(movieId)
+    }
+
+    override fun getAllMovie(callback: OnLoadDataCallback<MutableList<Movie>>) {
+        LocalAsyncTask<Unit, MutableList<Movie>>(callback) {
+            movieDao.getAllMovies()
+        }.execute(Unit)
```

Comment:
```suggestion
        }.execute()
```
Có Unit với không có Unit thì có gì khác nhau không ?